src/components/Button/Button.jsx

```diff
@@ -45,15 +51,20 @@ const Button = ({
   <button
     id={id}
     className={classNames(
-      buttonStyle(variant, modifier),
+      buttonStyle(variant, loading, modifier),
       buttonSize(size),
       className
     )}
     onClick={onClick}
     ref={innerRef}
+    disabled={loading}
+    aria-busy={loading}
     {...attrs}
   >
-    {children}
+    <span className="rvt-button__content" role="buttonContent">
```

Comment:
You could consider wrapping the button text in this span only if the loading prop is true, to mimic Rivet more closely.

Reply:
Good idea, thanks Rishi! I improved it.